source/common/unified_memory_allocation/include/uma/memory_pool.h

```diff
@@ -104,28 +104,23 @@ void umaPoolFree(uma_memory_pool_handle_t hPool, void *ptr);
 void umaFree(void *ptr);
 
 ///
-/// \brief Retrieve string representation of the underlying pool specific
-///        result reported by the last API that returned
-///        UMA_RESULT_ERROR_POOL_SPECIFIC or NULL ptr (in case of allocation
-///        functions). Allows for a pool independent way to
-///        return a pool specific result.
+/// \brief Retrieve uma_result_t representing the error of the last failed allocation
+///        operation in this thread (malloc, calloc, realloc, aligned_malloc).
 ///
 /// \details
-///     - The string returned via the ppMessage is a NULL terminated C style
-///       string.
-///     - The string returned via the ppMessage is thread local.
-///     - The memory in the string returned via the ppMessage is owned by the
-///       adapter.
-///     - The application may call this function from simultaneous threads.
-///     - The implementation of this function should be lock-free.
+/// * Implementations *must* store the error code in thread-local
+///   storage prior to returning NULL from the allocation functions.
+///
+/// * If the last allocation/de-allocation operation succeeded, the value returned by
+///   this function is unspecified.
+///
+/// * The application *may* call this function from simultaneous threads.
+///
+/// * The implementation of this function *should* be lock-free.
 /// \param hPool specified memory hPool
-/// \param ppMessage [out] pointer to a string containing provider specific
-///                  result in string representation.
-/// \return UMA_RESULT_SUCCESS if the result being
-///         reported is to be considered a warning. Any other result code
-///         returned indicates that the adapter specific result is an error.
-enum uma_result_t umaPoolGetLastResult(uma_memory_pool_handle_t hPool,
-                                       const char **ppMessage);
+/// \return UMA_RESULT_SUCCESS if the previous function completed successfully, error code
+///         desciribng the failure otherwise.
+enum uma_result_t umaPoolGetLastAllocationError(uma_memory_pool_handle_t hPool);
```

Comment:
Will it imply additional overhead? It means thread-local variable should be updated on every allocation.

Also, I am confused by the name. Do we restrict only to `Allocation` errors? What about other methods of the heap manager, e.g. `malloc_usable_size` or anything we can add in the future?

Reply:
Not on every allocation, just on allocation error - I just updated the comment, which incorrectly stated that the function can return OK if the previous allocation succeeded. The idea is that this function should only be called after a failed allocation function.

Yes, the reason for restricting it only to allocation errors is what I described in the PR description - for functions like free or malloc_usable_size we don't know if they failed so we cannot safely use this function. For other functions that we might add in the future, I assume they would be UMF-specific (not like allocation/free functions where we want to keep similar signature to libs functions) so we can just return UMF_STATUS